### ISSUE TITLE
Image resizing doesn't work for calendar entries

### ISSUE THREAD

**Issue by treasuretron** [spacebase/spacebasenz#208]
In GitLab by @richbodo on Jun 19, 2018, 18:07

URL: https://dev-54ta5gq-ensz4qdpjmziw.us.platform.sh/node/add/event

Expected: If an image is too large and is resized, it should be resized to within the size limitations

Observed: Images larger than alllowed are resized to be smaller than allowed.

![Screen_Shot_2018-06-19_at_2.55.03_PM](/uploads/5259c8a743500d7c160793f0dad907e3/Screen_Shot_2018-06-19_at_2.55.03_PM.png)

**Comment by treasuretron** [comment]
In GitLab by @richbodo on Jun 20, 2018, 01:05

Emeline noticed this as well and was having a hard time getting images up.

**Comment by treasuretron** [comment]
In GitLab by @137px on Jun 20, 2018, 09:51

@richbodo this is correct behavior, though in this instance it was a perfect storm that made it not work. If an image is too large in either dimension, it will resize it to fit within the constraints. If it's too small in either dimension, it won't upload.

In this case, the width is greater than the max width of 1100, so it resized the image to have a width of 1100, but then the height did not meet the minimum requirement of 480. 

The image should be 1100x500. I just set constraints to force that.

**Comment by treasuretron** [comment]
In GitLab by @137px on Jun 20, 2018, 09:58

I have not used anything with Drupal 8, but there were modules in 7 that allowed for any size image to be uploaded and then a cropping tool appears to let you crop it to the correct size on screen. That might be a better, more user friendly approach if the modules have been ported to D8.

**Comment by treasuretron** [comment]
In GitLab by @137px on Jun 21, 2018, 16:09

assigned to @richbodo and @Khiggins42

**Comment by treasuretron** [comment]
In GitLab by @137px on Jun 21, 2018, 16:09

@richbodo @Khiggins42  Image cropper is now on PR-173

Once this is approved and merged into dev, I will use this for the blog image as well.

**Comment by treasuretron** [comment]
In GitLab by @Khiggins42 on Jun 22, 2018, 15:23

@137px Looks good, the cropping feature works great with a sufficiently large starting image.

However, in an example image from one of Emeline's events, which is already a very wide image, there's no way to massage it into a useable image. I fully realize this is a matter of fitting a square peg into a round hole, and this may not be worth pursuing any further than this point. Any users entering events and hoping to repurpose an existing image from the event materials will just have to understand that. 

One suggestion is to allow crop AND resizing capability.

Em's example image:  

![Screen_Shot_2018-06-22_at_3.05.44_PM](/uploads/75918baf007ce1da2c900f15d828cfdd/Screen_Shot_2018-06-22_at_3.05.44_PM.png)

**Comment by treasuretron** [comment]
In GitLab by @richbodo on Jun 22, 2018, 16:03

Yeah, I think we'll need a resize module at some point.  I'll add that to the backlog and close this one out.  Thanks for the testing, kurt.

**Comment by treasuretron** [comment]
In GitLab by @richbodo on Jun 22, 2018, 16:03

closed
